lib/server.js

```diff
@@ -466,8 +468,9 @@ Server.prototype.pre = function pre() {
     var self = this;
     var handlers = Array.prototype.slice.call(arguments);
 
-    argumentsToChain(handlers).forEach(function forEach(h) {
-        self.before.push(h);
+    argumentsToChain(handlers).forEach(function forEach(handler) {
+        handler._name = handler.name || 'pre-' + self.preChain.count();
```

Comment:
Isn't this already happening in router?

Reply:
Router, pre, and use are registered separately with a different prefix.